### PR TITLE
Port DH-11185: Parallel Where

### PR DESCRIPTION
**Pull request by nbauernfeind**
This ports and enables the DHE implementation of parallel where.

It includes changes for these DHE tickets (links are internal):
- [DH-11185](https://deephaven.atlassian.net/browse/DH-11185): https://github.com/deephaven-ent/iris/commit/a080084c4c94d7025f0993c7de6fd28acea8ef2c
- [DH-11369](https://deephaven.atlassian.net/browse/DH-11369): https://github.com/deephaven-ent/iris/commit/3016275ee7b71f1881e399aacd6de018699332ff, and https://github.com/deephaven-ent/iris/commit/f82034afa571769b6f39f9f9098c3a2a6613813a
- [DH-11457](https://deephaven.atlassian.net/browse/DH-11457): https://github.com/deephaven-ent/iris/commit/b2fbc10061c7cd2a792b47017e15ff804258cef9

See nightly results [here](https://github.com/nbauernfeind/deephaven-core/actions/runs/4672351559).

### CODE REVIEW

File: engine/table/src/main/java/io/deephaven/engine/table/impl/WhereListener.java [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+        final TableUpdateImpl update = new TableUpdateImpl();
+
+        // intersect removed with pre-shift keyspace
+        update.removed = upstream.removed().intersect(currentMapping);
+        currentMapping.remove(update.removed);
+
+        // shift keyspace
+        upstream.shifted().apply(currentMapping);
+
+
```
Comment (lbooker42) [inline engine/table/src/main/java/io/deephaven/engine/table/impl/WhereListener.java:121]: Extra whitespace

File: engine/table/src/test/java/io/deephaven/engine/table/impl/QueryTableWhereTest.java [hunk 2]
```diff
@@ -269,23 +271,55 @@ public void testWhereInDependency() {
...
             TestCase.assertTrue(flushed);
 
+            // to get table 1 satisfied we need to still fire a notification for the filter execution, then the combined
+            // execution
+            if (QueryTable.FORCE_PARALLEL_WHERE) {
+                // the merged notification for table 2 goes first
+                flushed = UpdateGraphProcessor.DEFAULT.flushOneNotificationForUnitTests();
+                TestCase.assertTrue(flushed);
+
+                System.out.println("Flushing parallel notifications for setTable1");
```
Comment (lbooker42) [inline engine/table/src/test/java/io/deephaven/engine/table/impl/QueryTableWhereTest.java:281]: Are you wanting to keep these `println()` calls in the log or are they debug helpers?

**Review by lbooker42 — APPROVED** [review]
Looks good, very minor quibbles over naming and a single question.

File: engine/table/src/test/java/io/deephaven/engine/table/impl/QueryTableWhereTest.java (hunk 2 above)
Comment (nbauernfeind) [reply engine/table/src/test/java/io/deephaven/engine/table/impl/QueryTableWhereTest.java:281]: I turned these into log.debug invocations. I suspect they would be helpful if somehow parallel processing gets stuck and you'd like to better understand what those threads are supposed to be doing.

**Pull request merged**